block/manager.go

```diff
@@ -69,10 +67,10 @@ type Manager struct {
 	*/
 	// Protect against processing two blocks at once when there are two routines handling incoming gossiped blocks,
 	// and incoming DA blocks, respectively.
-	retrieverMu sync.Mutex
-	Retriever   da.BatchRetriever
-	// get the next target height to sync local state to
-	targetSyncHeight diodes.Diode
+	blockCacheMu sync.Mutex
+	// Protect against syncing twice from DA in case new batch is posted but it did not finish to sync yet.
```

Comment:
not sure I understand this. Seems to me like it's protecting against double entry to `syncToTargetHeight` but 1 of the 2 callers is `syncFromSettlement` which will always be called before the 2nd caller (as it's called on startup and blocking)

Reply:
if may happen syncToTargetHeight is not finished when there is a notification of a new batch and called again.